packages/platform/atoms/hooks/useGcal.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from "react";
+
+import http from "../lib/http";
+
+export interface useGcalProps {
+  isAuth: boolean;
+}
+
+export const useGcal = ({ isAuth }: useGcalProps) => {
+  const [allowConnect, setAllowConnect] = useState<boolean>(false);
+  const [checked, setChecked] = useState<boolean>(false);
+
+  const redirectToGcalOAuth = () => {
```

Comment:
All component functions should use `useEffect` to not be re-instantiated on each re-render.